actionpack/lib/action_dispatch/request/session.rb

```diff
@@ -163,7 +163,7 @@ def clear
       # Returns the session as Hash.
       def to_hash
         load_for_read!
-        @delegate.dup.delete_if { |_, v| v.nil? }
+        @delegate.dup.compact
```

Comment:
```suggestion
        @delegate.compact
```

since `compact` is returning a new hash anyway, we shouldn't need to `dup` as well